go/arrow/scalar/scalar.go

```diff
@@ -297,6 +298,8 @@ func (s *Decimal128) CastTo(to arrow.DataType) (Scalar, error) {
 	switch to.ID() {
 	case arrow.DECIMAL128:
 		return NewDecimal128Scalar(s.Value, to), nil
+	case arrow.DECIMAL256:
+		return NewDecimal256Scalar(decimal256.FromDecimal128(s.Value), to), nil
```

Comment:
Same as below: should take into account differences in scale or precision.

Reply:
fixed calling `Rescale` and `FitsInPrecision`